### DISCUSSION TITLE
Method chaining and commas in arrays

### DISCUSSION THREAD

**Discussion by julienfalque** [PHP-CS-Fixer/PHP-CS-Fixer#7877] [Feature ideas]
I sometimes use method chaining inside arrays. In such case, I move the comma on the next line for readability like I do with semicolons outside of arrays. When applying `no_whitespace_before_comma_in_array` rule, the comma is moved next to the last closing parenthesis:

```diff
 [
     $foo
         ->foo()
-        ->foo()
-    ,
+        ->foo(),
     $bar
         ->bar()
-        ->bar()
-    ,
+        ->bar(),
 ]
```

I would like the fixer to not move the comma when it is on a new line like in this example. Do you think it makes sense? If yes, do you consider this a bugfix or a feature request (so probably with a new configuration option to enable this behavior)?

**Comment by VincentLanglet** [comment]
I would expect this to be a feature request, it seems pretty similar to the `multiline_whitespace_before_semicolons` rule but with comma.
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/rules/semicolon/multiline_whitespace_before_semicolons.rst

There is a `new_line_for_chained_calls` in `multiline_whitespace_before_semicolons` rule.

**Reply by Wirone** [reply]
This _is_ feature request, but we slowly move to a new process. There's no point in having 7-years-old issue in the backlog, it's better to have ideas under Discussions and create issues only when there's an interest _and_ Definition of Done was crafted. The process is not established fully yet, though 😅.
